tests/unit/drtsans/test_transmission.py

```diff
@@ -280,5 +281,41 @@ def test_transmission(generic_workspace, clean_workspace):
     assert result.extractE()[0][0] == pytest.approx(0.005683898)  # expected transmission uncertainty
 
 
+@pytest.mark.parametrize(
+    "generic_workspace",
+    [
+        {
+            "name": "Isam",
+            "dx": pixel_size,
+            "dy": pixel_size,  # data requires a square pixel
+            "yc": pixel_size / 2.0,  # shift because the "detector" y-direction is even
+            "axis_values": [5.925, 6.075],
+            "intensities": Isam,
+        }
+    ],
+    indirect=True,
+)
+def test_transmission_error_tolerance(generic_workspace, clean_workspace):
+    """Test the transmission error tolerance exception"""
+    Isam = generic_workspace  # convenient name
+    clean_workspace(Isam)
+    assert Isam.extractY().sum() == 50212  # checksum
+
+    # generate the reference data - uncertainties are set separately
+    Iref = 1.8 * Isam
+    clean_workspace(Iref)
+    Iref = SetUncertainties(InputWorkspace=Iref, OutputWorkspace=Iref, SetError="sqrt")
+    assert Iref.extractY().sum() == 1.8 * 50212  # checksum
+    assert 1.8 * Isam.extractE().sum() > Iref.extractE().sum()  # shouldn't match
+
+    # run the algorithm
+    with pytest.raises(TransmissionErrorToleranceError) as exc_info:
+        calculate_transmission(Isam, Iref, 2.5 * pixel_size, "m", transmission_error_tolerance=0.01)
+
+    assert re.match(
+        "Transmission error 0.0102 > transmission error tolerance 0.0100 \(transmission 0.5556\)", str(exc_info.value)
```

Comment:
The message is a bit confusing because it compares and error to an error tolerance. How about
```
Transmission error (0.0102) > transmission * error_tolerance (0.5556 * 0.0100 = 0.0057)
```